test/threadstest_fips.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#if defined(_WIN32)
+# include <windows.h>
+#endif
+
+#include "testutil.h"
+#include "threadstest.h"
+
+static int success;
+
+static void thread_fips_rand_fetch(void)
+{
+    EVP_MD *md;
+
+    if (!TEST_true(md = EVP_MD_fetch(NULL, "SHA2-256", NULL)))
```

Comment:
We're using the default-and-fips.cnf file...so how do we know this comes from the fips provider? Should we have "fips=yes" here? Or use a different config file for this test?

Reply:
It doesn't matter, the provider has been loaded and initialised already.

I can make this an empty function if desirable.

Reply:
Surely the act of fetching loads the provider ? But yes - you're right it will get initialised regardless. So - ok.